client/src/components/TrackList.vue

```diff
@@ -141,11 +126,63 @@ export default Vue.extend({
         inputValue: item.checkedTrackIds.indexOf(item.track.trackId) >= 0,
         selected,
         editing: selected && item.editingTrack,
-        color: this.typeStyling.value.color(trackType),
+        color: typeStylingRef.value.color(trackType),
         types: item.allTypes,
-        frame: this.frame,
       };
-    },
+    }
+
+    watch(selectedTrackIdRef, scrollToTrack);
+    watch(tracksRef, scrollToSelectedTrack);
+
+    const mouseTrap = computed(() => {
+      const disabled = props.hotkeysDisabled;
+      return [
+        {
+          bind: 'up',
+          handler: (el: HTMLElement, event: KeyboardEvent) => {
+            scrollPreventDefault(el, event, 'up');
+          },
+          disabled,
+        },
+        {
+          bind: 'down',
+          handler: (el: HTMLElement, event: KeyboardEvent) => {
+            scrollPreventDefault(el, event, 'down');
+          },
+          disabled,
+        },
+        {
+          bind: 'enter',
+          handler: () => emit('track-click', selectedTrackIdRef.value),
```

Comment:
I have an idea about what to do about this to stop using `$emit` because of exactly this problem.  I want to expose the `hander` through the provide mechanism and let components call it directly rather than bubbling the event.

That would give us a huge boost in type safety and reduce complexity with no real drawback since we already took the singleton hit.

After that, we could start to build a global event system out of the provider to enable stuff like a dynamic help dialog.  We should discuss this at some point.

Anyway, I'll fix this here, thanks.